### PR TITLE
Supabase: improve sign in and sign up, add reset password

### PR DESCRIPTION
**Pull request by jordienr**
- Create a separate page for sign up
- UI improvements
- Add components to reduce tailwindpasta 
- Add reset password flow

### CODE REVIEW

File: examples/with-supabase/app/signup/page.tsx [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+import { Input } from "@/components/forms/input";
+import { Label } from "@/components/forms/label";
+import { FormMessage } from "@/components/forms/form-message";
+
+export default function Signup({
+  searchParams,
+}: {
+  searchParams: FormMessage;
+}) {
+  const signUp = async (formData: FormData) => {
```
Comment (leerob) [inline examples/with-supabase/app/signup/page.tsx:15]: This could probably be moved to an `actions.ts` file

**Review by leerob — APPROVED** [review]

File: examples/with-supabase/app/login/page.tsx [hunk 2]
```diff
@@ -22,37 +20,14 @@ export default function Login({
     });
 
     if (error) {
-      return redirect("/login?message=Could not authenticate user");
+      return redirect("/login?error=Could not authenticate user");
```
Comment (hf) [inline examples/with-supabase/app/login/page.tsx:23]: This should be URL encoded please.

File: examples/with-supabase/app/protected/reset-password/page.tsx [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+  const resetPassword = async (formData: FormData) => {
+    "use server";
+    const supabase = createClient();
+
+    const password = formData.get("password") as string;
+    const confirmPassword = formData.get("confirmPassword") as string;
+
+    if (!password || !confirmPassword) {
+      redirect(
+        "/protected/reset-password?error=Password and confirm password are required",
```
Comment (hf) [inline examples/with-supabase/app/protected/reset-password/page.tsx:22]: URL encoding here too.

File: examples/with-supabase/app/protected/reset-password/page.tsx [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+      );
+    }
+
+    if (password !== confirmPassword) {
+      redirect("/protected/reset-password?error=Passwords do not match");
```
Comment (hf) [inline examples/with-supabase/app/protected/reset-password/page.tsx:27]: And here.

File: examples/with-supabase/app/login/page.tsx (hunk 2 above)
Comment (jordienr) [reply examples/with-supabase/app/login/page.tsx]: fixed thx!

**Pull request merged**